src/structured-text/structured_text_language.cpp

```diff
@@ -0,0 +1,131 @@
+/*******************************************************************\
+
+Module: Structured Text Language Interface
+
+Author: Matthias Weiss, matthias.weiss@diffblue.com
+
+\*******************************************************************/
+
+/// \file
+/// Structured Text Language Interface
+
+#include "structured_text_language.h"
+#include "converters/expr2structured_text.h"
+#include "structured_text_parser.h"
+#include "structured_text_typecheck.h"
+
+#include <linking/linking.h>
+#include <linking/remove_internal_symbols.h>
+#include <util/get_base_name.h>
+
+bool structured_text_languaget::generate_support_functions(
+  symbol_tablet &symbol_table)
+{
+  return false;
+}
+
+bool structured_text_languaget::typecheck(
+  symbol_tablet &symbol_table,
+  const std::string &module,
+  const bool keep_file_local)
+{
+  symbol_tablet new_symbol_table;
+
+  if(structured_text_typecheck(
+       parse_tree, new_symbol_table, module, get_message_handler()))
+  {
+    return true;
+  }
+
+  remove_internal_symbols(
+    new_symbol_table, this->get_message_handler(), keep_file_local);
+
+  if(linking(symbol_table, new_symbol_table, get_message_handler()))
+    return true;
+
+  return false;
+}
+
+bool structured_text_languaget::parse(
+  std::istream &instream,
+  const std::string &path)
+{
+  structured_text_parser.clear();
+  parse_path = path;
+  structured_text_parser.set_line_no(0);
+  structured_text_parser.set_file(path);
+  structured_text_parser.in = &instream;
+  structured_text_scanner_init();
+  bool result = structured_text_parser.parse();
+
+  // store result
+  structured_text_parser.swap_tree(parse_tree);
+
+  return result;
+}
+
+void structured_text_languaget::show_parse(std::ostream &out)
+{
+  parse_tree.output(out);
+}
+
+bool structured_text_languaget::from_expr(
+  const exprt &expr,
+  std::string &code,
+  const namespacet &ns)
+{
+  code = expr2stl(expr);
+  return false;
+}
+
+bool structured_text_languaget::from_type(
+  const typet &type,
+  std::string &code,
+  const namespacet &ns)
+{
+  code = type2stl(type);
+  return false;
+}
+
+bool structured_text_languaget::type_to_name(
+  const typet &type,
+  std::string &name,
+  const namespacet &ns)
+{
+  return from_type(type, name, ns);
+}
+
+bool structured_text_languaget::to_expr(
+  const std::string &code,
+  const std::string &module,
+  exprt &expr,
+  const namespacet &ns)
+{
+  // unused parameters
```

Comment:
Is this to avoid compiler warnings?

Reply:
Yes, although I don't know if this is necessary anymore (`ansi_c_languaget` does it in a similar way at least).

Reply:
Would it work to not name them instead (`to_expr(const std::string &, const std::string &, ...)`)?